sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala

```diff
@@ -627,4 +627,19 @@ class AnalysisSuite extends AnalysisTest with Matchers {
     val res = ViewAnalyzer.execute(view)
     comparePlans(res, expected)
   }
+
+  test("SPARK-26741: replace aggregate expressions in Sort when Aggregate is not a direct child") {
```

Comment:
Can we have end-to-end test case other than this one? Or the test case in `SubquerySuite` is the only one to produce this issue?

Reply:
I am not sure which other end-to-end test case you are thinking to. So far, the only case when I saw this happening is when there is a HAVING clause. So I am not sure which other end-to-end tests you want to add.

Reply:
Ok, I'm not sure too. Just want to make sure if this is the only case we are aware of.

Reply:
yes, this is the only case I am aware of.